homeassistant/components/device_tracker/__init__.py

```diff
@@ -486,7 +486,7 @@ def get_vendor_for_mac(self):
             return None
 
         # prevent lookup of invalid macs
-        if not len(self.mac.split(':')) == 6:
+        if not len(self.mac) == 17:
```

Comment:
Because MAC addresses are not normalized it can happen a address contains a single `0` instead of `00`. This change would break the vendor lookup in those cases. It would be better imho to catch and pass the error silently at this point: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/device_tracker/__init__.py#L496